pennylane/math/quantum.py

```diff
@@ -281,15 +280,17 @@ def _batched_partial_trace(density_matrix, indices):
     """
     # Autograd does not support same indices sum in backprop, and tensorflow
     # has a limit of 8 dimensions if same indices are used
-    if get_interface(density_matrix) in ["autograd", "tensorflow"]:
-        return _batched_partial_trace_nonrep_indices(density_matrix, indices)
+    matrix = cast(matrix, dtype=c_dtype)
+
+    if get_interface(matrix) in ["autograd", "tensorflow"]:
+        return _batched_partial_trace_nonrep_indices(matrix, indices)
 
     # Dimension and reshape

```

Comment:
Before this, you could add a small check to allow for simply computing the partial trace of a single matrix (with shape `(2**n, 2**n)`. I.e.

```
if qml.math.ndim(matrix)==2:
    batch_dim = 1
    dim = len(matrix)
else:
    batch_dim, dim = matrix.shape[:2]
```

I _think_ this should already be sufficient since below there is a reshaping anyway. You may have to play with this a little and also make sure to check the `_batched_partial_trace_nonrep_indices` branch.
You should then add tests accordingly, confirming that passing a single matrix acts as expected.

Reply:
added this to both partial_trace and _batched_partial_trace_nonrep_indices, which I believe fixed the issue

Reply:
brings up another issue. The unit tests for density matrices fail when adding this condition in. Currently working on debugging.